Tree: Implement Forest proxy (#11589)
microsoft/FluidFramework: packages/dds/tree/src/feature-libraries/editable-tree/README.md

> # Editable Tree

Reviewer: I recently normalized all these sub-remes to just match the directory/package name with their title exactly.
```suggestion
# editable-tree
```

# Editable Tree

> Editable Tree is a simple API prividing an access to a Forest data.

Reviewer: ```suggestion
Editable Tree is a simple API for accessing a Forest's data.
```

> An entry point is a `getEditableTree` function which returns a JS Proxy typed as `IEditableTree` allowing to read and write nodes of the Fores in a "JavaScript-object"-like manner, thus hiding all the Forest complexity.
> 
> This is intended to showcase basic read and write opeartions from and to the Forest based on a reference `ObjectForest` and a corresponding `Cursor` implementation.
> 
> Write operations require a schema-based data format convertion to be adopted/implemented, which will follow.

Reviewer: I think the use of proxies is an implementation detail, and it would be nice to explicitly separate the goals of this library from its internals and implementation status. Also nothing in here should be ObjectForest specific.

```suggestion
The entry point is `getEditableTree` which returns an `IEditableTree` which allows reading and writing nodes of the Forest in a "JavaScript-object"-like manner.

## Status

Currently implementation uses a JavaScript proxy and is focused on reading the tree: support for editing is not yet implemented.

Write operations will require a schema-based data format conversion to be implemented.

```
